docs/DeveloperGuide.md

```diff
@@ -896,85 +916,69 @@ After which, a new `SortCommand` object will be created, and is subsequently exe
+#### Implementation
 
-![UndoRedoState5](images/UndoRedoState5.png)
+1. When the user launches the application, `ModelManager` is initialised with a new `AddressBookHistory` object.
+2. Whenever the user successfully executes a command (excluding `clear` and `undo` itself), the `LogicManager` calls `Model#saveCurrentAddressBookToHistory()`. This adds a copy of TAPA to the list of TAPA’s previous states in `AddressBookHistory`.
 
-The following activity diagram summarizes what happens when a user executes a new command:
+![AddToAddressBookHistorySequenceDiagram](images/AddToAddressBookHistorySequenceDiagram.png)
 
-<img src="images/CommitActivityDiagram.png" width="250" />
+4. When the user inputs `undo`, the `LogicManager` parses the given input text using `AddressBookParser#parseCommand()`.
+5. A new `UndoCommand` is created by `AddressBookParser` and returned to the `LogicManager`.
+6. The `LogicManager` will then call `UndoCommand#execute(Model model)`.
+7. Following this, the `UndoCommand` calls `Model#undoAddressBook()` to revert the current details stored in TAPA to those stored before the previous command was executed. (Within `ModelManager`, `AddressBookHistory#getPreviousAddressBook` is called to retrieve the state of TAPA before the previous command.)
+8. Lastly, the `UndoCommand` creates a new `CommandResult` which is returned to the `LogicManager`.
 
-#### Design considerations:
+![UndoCommandSequenceDiagram](images/UndoCommandSequenceDiagram.png)
 
-**Aspect: How undo & redo executes:**
+#### Design considerations
 
-* **Alternative 1 (current choice):** Saves the entire address book.
-  * Pros: Easy to implement.
-  * Cons: May have performance issues in terms of memory usage.
+**Aspect: How the `UndoCommand` reverts the changes made by the previous command**
 
-* **Alternative 2:** Individual command knows how to undo/redo by
-  itself.
-  * Pros: Will use less memory (e.g. for `delete`, just save the person being deleted).
-  * Cons: We must ensure that the implementation of each individual command are correct.
+* **Alternative 1 (current choice):** Store the details of TAPA after each successfully executed command, then recover them when `undo` is executed.
+   * Pros: Easy to implement and test.
+   * Cons: Will take up more memory with each command executed.
 
-_{more aspects and alternatives to be added}_
+* **Alternative 2:** Extend each command to be able to revert the changes it has made to TAPA.
+   * Pros: Will not incur major performance issues as it uses less memory.
+   * Cons: More difficult to implement and test. Each command would need a unique implementation to be undone and this would also need to be implemented for commands added in the future.
+   
+*{More features to be added}*
```

Comment:
Should we remove this since we have no other features? I think it's perfectly fine if we were to leave it as it is too.
```suggestion
```